sample_projects/LEQuad/lequad.cpp

```diff
@@ -0,0 +1,629 @@
+    state_machine(state, position_estimation, imu, ahrs, manual_control),
+    data_logging_continuous(file1, state, config.data_logging_continuous_config),
+    data_logging_stat(file2, state, config.data_logging_stat_config),
+    sysid_(mavlink_communication.sysid()),
+    config_(config)
+{
+    // Init al modules
+    init_state();
+    init_communication();
+    init_data_logging();
+    init_gps();
+    init_imu();
+    init_barometer();
+    init_sonar();
+    init_attitude_estimation();
+    init_position_estimation();
+    init_stabilisers();
+    init_navigation();
+    init_hud();
+    init_servos();
+    init_ground_control();
+}
+
+
+void LEQuad::loop(void)
+{
+    // Sort tasks
+    scheduler.sort_tasks();
+    mavlink_communication.scheduler().sort_tasks();
+
+    // Try to read from flash, if unsuccessful, write to flash
+    if (mavlink_communication.onboard_parameters().read_parameters_from_storage() == false)
+    {
+        mavlink_communication.onboard_parameters().write_parameters_to_storage();
+    }
+
+    // Create log files
+    ret &= data_logging_continuous.create_new_log_file("Log_file", true, mavlink_communication.sysid());
+    ret &= data_logging_stat.create_new_log_file("Log_Stat", false, mavlink_communication.sysid());
+    
+    // Init mav state
+    state.mav_state_ = MAV_STATE_STANDBY;  // TODO check if this is necessary
+
+    while (1)
+    {
+        scheduler.update();
+    }
+}
+
+
+// -------------------------------------------------------------------------
+// State
+// -------------------------------------------------------------------------
+bool LEQuad::init_state(void)
+{
+    bool ret = true;
+
+    // UP telemetry
+    ret &= state_telemetry_init(&state, mavlink_communication.p_message_handler());
+
+    // DOWN telemetry
+    ret &= mavlink_communication.add_msg_send(MAVLINK_MSG_ID_HEARTBEAT,  1000000, (Mavlink_communication::send_msg_function_t)&state_telemetry_send_heartbeat, &state);
+    ret &= mavlink_communication.add_msg_send(MAVLINK_MSG_ID_SYS_STATUS, 1000000, (Mavlink_communication::send_msg_function_t)&state_telemetry_send_status,    &state);
+
+    // Data logging
+    ret &= data_logging_stat.add_field((uint32_t*)&state.mav_state_, "mav_state");
+    ret &= data_logging_stat.add_field(&state.mav_mode_,             "mav_mode");
+
+    // Task
+    ret &= scheduler.add_task(200000, (Scheduler_task::task_function_t)&State_machine::update, (Scheduler_task::task_argument_t)&state_machine);
+    ret &= scheduler.add_task(500000, (Scheduler_task::task_function_t)&task_led_toggle,       (Scheduler_task::task_argument_t)&led, Scheduler_task::PRIORITY_LOW);
+
+    return ret;
+}
+
+
+// -------------------------------------------------------------------------
+// Communication
+// -------------------------------------------------------------------------
+bool LEQuad::init_communication(void)
+{
+    bool ret = true;
+
```

Comment:
Don't you want to keep the same structure and leave it empty when no up stream... ?